Makefile

```diff
@@ -0,0 +1,158 @@
+VIRTUALENV=~/.venvs/pixua
+
+.PHONY: setup
+setup: requirements install
+
+.PHONY: requirements
+requirements: requirements_py requirements_js
+	@echo "Installed requirements"
+
+.PHONY: install
+install: requirements install_py install_js
+	@echo "Installed dependencies"
+
+.PHONY: requirements_py
+requirements_py: create_virtualenv
+	@echo "Upgrading pip and setuptools"
+	@source $(VIRTUALENV)/bin/activate;\
+	pip install --upgrade pip setuptools
+
+.PHONY: install_py
+install_py: service/requirements.txt service/test_requirements.txt
+	@echo "Installing python packages"
+	@source $(VIRTUALENV)/bin/activate;\
+	cd service;\
+	pip install --exists-action s -r requirements.txt -r test_requirements.txt
+
+.PHONY: requirements_js
+requirements_js:
+	@echo "Installing javascript npm and bower dependencies"
+	@cd web-ui;\
+	npm install;\
+	node_modules/.bin/bower install
+
+.PHONY: install_js
+install_js:
+	@echo "Building front-end and static files"
+	@cd web-ui;\
+	npm run build
+
+.PHONY: create_virtualenv
+create_virtualenv: ensure_virtualenv_installed
+	@if [ ! -e $(VIRTUALENV)/bin/activate ]; then\
+		echo "Pixelated virtualenv doesn't exist, creating now";\
+		virtualenv --python=python2 $(VIRTUALENV);\
+	else\
+    echo "Pixelated virtualenv already exists, moving on";\
+	fi
+
+.PHONY: test
+test: test_py test_js coverage
+
+.PHONY: test_py
+test_py: clean requirements install coverage linters_py unit_tests_py integration_tests_py
+
+.PHONY: test_js
+test_js: clean requirements_js install_js linters_js unit_tests_js
```

Comment:
Again, both depend on clean, but if you run test (that depend on test_js and test_py), clean will be called only once. It doesn't hurt to have dependencies duplicated if they are indeed needed by both targets because make will solve that in it's tree and run it only once.